GenericAuthControl - authenticate without need to create custom Control

MartinMystikJonas opened the pull request.

f3l1x commented on composer.json:
```diff
@@ -21,6 +21,7 @@
     "nette/http": "^3.0.5"
   },
   "require-dev": {
+    "nette/application": "^v3.0.0",
```
```suggestion
    "nette/application": "^3.0.0",
```

MartinMystikJonas commented:
Dependencies fixed, rebased on master and fixed BC with older Nette packages used with --prefer-lowest (Arrays::invoke is not available till nette/utils 3.2.1)

Now it should be ready to merge.

f3l1x commented on tests/fixtures/Presenter/TestPresenter.php:
```diff
@@ -0,0 +1,51 @@
+	/** @var Component */
+	private $component;
+
+	/** @var Request */
```
HttpRequest

f3l1x commented on tests/fixtures/Presenter/TestPresenter.php:
```diff
@@ -0,0 +1,51 @@
+	/** @var Request */
+	public $httpRequest;
+
+	/** @var Response */
```
HttpRequest alias

f3l1x commented on tests/fixtures/Presenter/TestPresenter.php:
```diff
@@ -0,0 +1,51 @@
+	/** @var Response */
+	public $httpResponse;
+
+	/**	@var \Nette\Application\Response */
```
FQN

f3l1x commented on tests/fixtures/Presenter/TestPresenter.php:
```diff
@@ -0,0 +1,51 @@
+		return $this->component;
+	}
+
+	public function sendResponse(\Nette\Application\Response $response): void
```
FQN

f3l1x commented on src/UI/Components/GenericAuthControl.php:
```diff
@@ -0,0 +1,92 @@
+  /** @var AuthCodeFlow */
+	private $authCodeFlow;
+
+  /** @var string|null */
```
Formatting
